lib/buffer.js

```diff
@@ -107,7 +111,7 @@ function Buffer(arg, encodingOrOffset, length) {
         'If encoding is specified then the first argument must be a string'
       );
     }
-    return Buffer.allocUnsafe(arg);
+    return allocate(arg);
```

Comment:
Just wondering - wouldn't be better to put negative check into `allocate` so that under any conditions it would be checked & impossible to create negative buffer? Unlike `typeof === 'number'`, I don't see cases where this shouldn't be applied.